### PR TITLE
example-frameworks: add a prometheus example

### PR DESCRIPTION
**Pull request by nicks**
Hello @landism, @maiamcc,

Please review the following commits I made in branch nicks/prometheus:

4f437160e36cf084b7aaba96b5471da45f60406a (2020-08-25 19:36:13 -0400)
example-frameworks: add a prometheus example

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
I like the general pattern here of "extract CRDs and use a local resource to wait for them to come up", though not sure yet if there's a good way to move this into an extension. helm_remote does something like this but it's not quite as clean as here.

File: prometheus/Tiltfile [hunk 1]
```diff
@@ -0,0 +1,61 @@
+
+# Download the manifest and apply it.
+bundle = local("curl -sSL https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.41.1/bundle.yaml", quiet=True)
```
Comment (landism) [inline prometheus/Tiltfile]:
```suggestion
bundle = local("curl -fsSL https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.41.1/bundle.yaml", quiet=True)
```

without `-f`, curl will exit with status 0 even on 4xx/5xx

File: prometheus/cluster-role-binding.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: prometheus
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: prometheus
+subjects:
```
Comment (landism) [inline prometheus/cluster-role-binding.yaml:9]: I got this error on `tilt up`:
```
Build Failed: kubectl apply:
stderr: The ClusterRoleBinding "prometheus" is invalid: subjects[0].namespace: Required value
: exit status 1
```

which went away when I added a `namespace: default` here

**Review by landism — COMMENTED** [review]
I pretty consistently get a PodRestartError alert on the 'prometheus' resource when I `tilt up`. Have you seen this?

https://cloud.tilt.dev/snapshot/AfSGtPQLT7qlSm6GqBA=

File: prometheus/Tiltfile (hunk 1 above)
Comment (nicks) [reply prometheus/Tiltfile]: yes! added

File: prometheus/cluster-role-binding.yaml (hunk 2 above)
Comment (nicks) [reply prometheus/cluster-role-binding.yaml:9]: oops, you're right, i messed that up when i was cleaning up the pr for general use

**Comment by nicks** [comment]
re: the alert - ya, that's just an artifact of how the prometheus operator works -- it isn't able to generate the config until after the first restart. i left a note about it in the readme

**Pull request merged**